File: algo/src/main/java/com/hartwig/actin/algo/evaluation/washout/WillRequireAnticancerTherapy.java
package com.hartwig.actin.algo.evaluation.washout;

import com.hartwig.actin.PatientRecord;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import com.hartwig.actin.algo.evaluation.EvaluationFactory;
import com.hartwig.actin.algo.evaluation.EvaluationFunction;

import org.jetbrains.annotations.NotNull;

public class WillRequireAnticancerTherapy implements EvaluationFunction {

    WillRequireAnticancerTherapy() {
    }

    @NotNull
    @Override
    public Evaluation evaluate(@NotNull PatientRecord record) {
        return EvaluationFactory.create(EvaluationResult.NOT_EVALUATED);
    }
}

File: algo/src/main/kotlin/com/hartwig/actin/algo/evaluation/washout/WillRequireAnticancerTherapy.kt
package com.hartwig.actin.algo.evaluation.washout

import com.hartwig.actin.PatientRecord
import com.hartwig.actin.algo.datamodel.Evaluation
import com.hartwig.actin.algo.evaluation.EvaluationFactory
import com.hartwig.actin.algo.evaluation.EvaluationFunction

class WillRequireAnticancerTherapy internal constructor() : EvaluationFunction {
    override fun evaluate(record: PatientRecord): Evaluation {
        return EvaluationFactory.notEvaluated(
            "Currently not evaluated whether anticancer therapy is required",
            "Anticancer therapy requirement not assumed"
        )
    }
}